internal/cmd/stop.go

```diff
@@ -17,15 +21,53 @@ func SetupStopCli() *cobra.Command {
 		Short: "Stop CNTLM Proxy",
 		Run:   stopCmd,
 	}
+	stopCmd.Flags().Bool("all", true, "unset all proxy configuration")
 	return stopCmd
 }
 
 func stopCmd(cmd *cobra.Command, args []string) {
-	profile.UpdateGlobalEnvironmentVariables("")
+	cfg, err := config.LoadConfiguration(proxyProfile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if cmd.Flags().Changed("all") {
+		if err = unsetAllConfiguration(&cfg); err != nil {
+			log.Fatal(err)
+		}
+	}
+	emptyURL, err := url.Parse("")
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = env.UpdateGlobalEnvironmentVariables(emptyURL)
+	if err != nil {
+		log.Fatal(err)
+	}
 	cmds := execute.Command{Cmd: "pkill", Args: []string{"cntlm"}}
-	_, err := execute.RunCommand(cmds)
+	_, err = execute.RunCommand(cmds)
```

Comment:
This assumes pkill is available on the machine that go-proxy is run from. If we store the PID of the CNTLM process instead of `Running: true` (as mentioned above) we can use https://golang.org/pkg/os/#FindProcess and Kill to kill it in a cross-platform (and more reliable way). E.g. this will kill cntlm even if it was not started by the go-proxy tool.